__tests__/tests/components/CommentInput.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {Platform} from 'react-native';
+
+import {CommentInput} from 'components';
+
+describe('<CommentInput />', () => {
+
+  let defaultProps = {
+    users: [],
+    userHasPushPermission: true,
+    issueLocked: false,
+    locale: '',
+    onSubmit: () => {},
+  };
+
+  it("should render 'TextInput' and TouchableOpacity if user can post", () => {
+
+    const wrapper = shallow(
+      <CommentInput {...defaultProps}/>
+    );
+
+    expect(wrapper.find('TextInput').length).toEqual(1);
+    expect(wrapper.find('TouchableOpacity').length).toEqual(1);
+    expect(wrapper.find('Text').length).toEqual(0);
+  });
+
+  it("should not render 'TextInput' and TouchableOpacity if user can post", () => {
+
+    const wrapper = shallow(
+      <CommentInput
+        {...defaultProps}
+        userHasPushPermission={false}
+        issueLocked={true}/>
+    );
+
+    expect(wrapper.find('TextInput').length).toEqual(0);
+    expect(wrapper.find('TouchableOpacity').length).toEqual(0);
+    expect(wrapper.find('Text').length).toEqual(1);
+  });
+
+  it('should update the state text if value change', () => {
+    const wrapper = shallow(
+      <CommentInput
+        {...defaultProps}
+        userHasPushPermission={true}
+        issueLocked={false}/>
+    );
+
+    const input = wrapper.find('TextInput');
+
+    input.simulate('changeText', 'Changed text');
+
+    expect(wrapper.state('text')).toEqual('Changed text');
+  });
+
+  it('should call handleSubmit methods when submitted', () => {
+    const wrapper = shallow(
+      <CommentInput
+        {...defaultProps}
+        userHasPushPermission={true}
+        issueLocked={false}/>
+    );
+
+    const handleSubmitSpy = jest.spyOn(wrapper.instance(), 'handleSubmit');
+
+    wrapper
+      .instance()
+      .forceUpdate();
+
+    wrapper
+      .find('TextInput')
+      .simulate('changeText', 'Changed text');
+
+    wrapper
+      .find('TouchableOpacity')
+      .simulate('press');
+
+    expect(handleSubmitSpy).toHaveBeenCalled();
+  });
+
+  it('should change the content size', () => {
+    const wrapper = shallow(
+      <CommentInput
+        {...defaultProps}
+        userHasPushPermission={true}
+        issueLocked={false}/>
+    );
+
+    wrapper
+      .find('TextInput')
+      .simulate('contentSizeChange', {
+        nativeEvent: {
+          contentSize: {
+            height: 10
+          }
+        }
+      });
+
+    expect(wrapper.state('height')).toBe(10);
+  });
+
+  it('should call handleSubmitEditing methods when onSubmitEditing event raised', () => {
+    const wrapper = shallow(
+      <CommentInput
+        {...defaultProps}
+        userHasPushPermission={true}
+        issueLocked={false}/>
+    );
+
+    const handleSubmitEditingSpy = jest.spyOn(wrapper.instance(), 'handleSubmitEditing');
+
+    wrapper
+      .instance()
+      .forceUpdate();
+
+    Platform.OS = 'android';
```

Comment:
Can you explain this line for me? I am curious about `iOS`.

Reply:
There is an instruction for the platform Android in the method `handleSubmitEditing`. And the `Platform` object in imported in the class `CommentInput` from `'react-native'`. But I failed to mock and import the platform object in `CommentInput`. I can use `inject-loader` as explained in [this article](https://www.jamestease.co.uk/blether/mocking-es6-imports-for-tests) but I have to install it.
![image](https://user-images.githubusercontent.com/5528931/32132895-510aa0e0-bbc4-11e7-8773-fec8e70596d3.png)

About `iOS`; there is no condition for that in  `CommentInput` class